src/ngraph/pass/concat_fusion.cpp

```diff
@@ -99,22 +99,31 @@ void pass::ConcatElimination::construct_concat_elimination()
             << "concat_elimination: In callback for construct_concat_elimination against node = "
             << m.get_match_root()->get_name();
         auto pattern_map = m.get_pattern_map();
-        auto op = pattern_map[op_label];
+        auto concat_input_op = pattern_map[op_label];
 
-        auto root = as_type_ptr<op::Concat>(m.get_match_root());
-        if (root && (root->get_input_shape(0) == root->get_output_shape(0)))
+        auto concat = as_type_ptr<op::Concat>(m.get_match_root());
+        // eliminate concat with 1 input
+        if (concat && (concat->get_input_size() == 1))
         {
-            NGRAPH_DEBUG << " eliminated " << m.get_match_root() << "\n";
-            replace_node(m.get_match_root(), op);
-
+            NGRAPH_DEBUG << " eliminated " << concat << "\n";
+            concat_input_op->set_friendly_name(concat->get_friendly_name());
+            replace_node(concat, concat_input_op);
             return true;
         }
         NGRAPH_DEBUG << " Incorrect match in callback\n";
         return false;
     };
 
     auto m = std::make_shared<pattern::Matcher>(concat_label, "ConcatElimination");
-    this->add_matcher(m, callback, PassProperty::REQUIRE_STATIC_SHAPE);
+    this->add_matcher(m, callback);
+}
+
+bool ngraph::pass::ConcatElimination::run_on_function(std::shared_ptr<Function> function)
+{
```

Comment:
What problem is this working around?

Reply:
they have CI / functional tests with the graph having just concat op node (with single input in some cases). This pass was eliminates even these, resulting graph "parameter->result" (no op node). This fix avoids the pass if the graph is just one op node.